### DISCUSSION TITLE
Yuzu

### DISCUSSION THREAD

**Discussion by john30** [john30/ebusd#1368] [Announcements]
This is the 25th (="Y", therefor [Yuzu](https://en.wikipedia.org/wiki/Yuzu)) release of ebusd since it was completely reworked.

The major new features are mDNS-based device auto-discovery, the switch of default config path to [ebus.github.io](https://ebus.github.io/) generated from TypeSpec sources, additional commands for device simulation and testing, as well as a bunch of bug fixes.

See [ChangeLog](https://github.com/john30/ebusd/blob/master/ChangeLog.md) and [Wiki](https://github.com/john30/ebusd/wiki) for details.

Debian 12 (Bookworm), 11 (Bullseye), and 10 (Buster) install packages for amd64, 386, armv7 (Raspberry Pi 2/3/4), and arm64 (Raspberry Pi 3/4 64 bit) are attached and will also be made available soon in the [Debian repository as described here](https://github.com/john30/ebusd-debian/blob/master/README.md).

For convenience, an armv6l (Raspberry Pi 1 and Zero/Zero W) Buster image is attached here as well (not part of the Debian repository though).

Variants of each binary with MQTT support have an additional "mqtt1" (for libmosquitto1) suffix in the name, whereas those without don't.

For Arch Linux, simply run `makepkg` in the `contrib/archlinux/ebusd/` directory.

The latest release is also available as [multi-architecture Docker image on the hub](https://hub.docker.com/r/john30/ebusd) with tag ebusd:latest.

<hr /><em>This discussion was created from the release <a href='https://github.com/john30/ebusd/releases/tag/24.1'>Yuzu</a>.</em>

**Comment by hemagi** [comment]
Hi John, due to mDNS, is is safe to upgrade 23.2 to 24.1 without changing the configuration?

I am using a viallant heat pump, and `ebusctl i` states that the CSVs have changed:

```
update check: version 24.1 available, broadcast.csv: newer version available, vaillant/08.ehp.csv: newer version available, vaillant/15.uih.csv: newer version available, vaillant/timer.inc: newer version available, vaillant/yield8485.inc: newer version available
```

Is this due to change of location of the config files, or has their content changed?

**Reply by john30** [reply]
mdns is just a new option, so of course you can still assign the device string manually.

for the second question, see the [changelog](https://github.com/john30/ebusd-configuration/blob/master/ChangeLog.md)

**Comment by Mozzarella7** [comment]
Hi John, 
is it possible to keep TypeSpec source file locally as in "old" version using **EBUSD_CONFIGPATH** parameter. 
If yes I understand that all I need is in **/master/src** catalog.

Thank you for the continuous development of this project!

**Reply by john30** [reply]
the local use of newly generated csv from typespec is described in the [readme](https://github.com/john30/ebusd-configuration#using-the-cdn)

**Comment by leonundjulie** [comment]
Hi John. However, I noticed the available update to the actual one 24.1 not before :-( About this I Need to know how the update procedure  hast to be performed? Do I have to uninstall the previous version first? Or can I use the new version as with the first installation, i.e. simply overwrite the first version?

**Reply by john30** [reply]
that depends entirely on the way of using it, i.e. docker, local install, managed HA addone etc.
for debian package it works just like any other, for docker a pull and recreate of the container is enough

**Reply by leonundjulie** [reply]
@john30 , my mistake … I am operating a Raspberry 4 with 64 Bit bookworm OS

**Comment by leonundjulie** [comment]
/ @john30 , what is going wrong. I am operating a RASPBERRY PI 4 with a 64Bit bookworm OS. What is my mistake Düring my Installation?

The Photo Shows the Data of my PI:
<img width="706" alt="Bildschirmfoto 2025-01-11 um 08 00 38" src="https://github.com/user-attachments/assets/78fdcaac-8e17-4785-90ee-b55d61abdf00" />

This Photo Show what happened during the Installation process:
![Bild 13 01 25 um 18 59](https://github.com/user-attachments/assets/509d5f2d-9594-4457-b317-cd6b8dc4fdec)

**Reply by john30** [reply]
looks as if the URL you used is invalid

**Reply by leonundjulie** [reply]
Thanks for your Replikation. And indead, I made copy mistake. However, I can Not Check before today evening due to limited possibilities to connect my PI remotely.

**Reply by leonundjulie** [reply]
Thanks, so far everything is working

**Comment by Pamirkr** [comment]
Hi John, I have a problem with running this version of ebusd with mqtt user's password option.  I tried starting ebus as a service and input password in /etc/default/ebus file.  Also I tried running ebusd directly from CLI:

`root@wyse:~# ebusd --mqtthost=192.168.1.101 --mqttport=1883 --mqttint=/etc/ebusd/mqtt-hassio.cfg --mqttjson --mqttuser=ebus --mqttpass=ebus`

But result is the same. Null password is sent to MQTT broker and connection is refused. Any idea where I make a mistake?

Ebus configuration file:
![4](https://github.com/user-attachments/assets/f1921d4e-9d85-4e19-a406-e61faaef7178)

Ebusd service status:
![1](https://github.com/user-attachments/assets/f982c019-45c4-436b-89b5-9f35276904c0)

Ebusd log:
![2](https://github.com/user-attachments/assets/bffc377e-04de-4c65-ad89-46f6a7c10b9e)

Log from Mosquitto broker:
![3](https://github.com/user-attachments/assets/4343d866-dbfc-471d-afc1-bedffa5f061a)

**Reply by Koky05** [reply]
Did you create that `ebus` user under MQTT broker?

**Comment by Pamirkr** [comment]
Yes, I did. I also found the cause of the problem. I did not notice that there were other ebusd processes running in the background without user and password set for MQTT. After killing them connection with mqtt broker works.
